src/java/org/apache/cassandra/index/sai/StorageAttachedIndex.java

```diff
@@ -317,6 +321,27 @@ public static Map<String, String> validateOptions(Map<String, String> options, T
         if (duplicateCount > 1)
             throw new InvalidRequestException(String.format("Cannot create duplicate storage-attached index on column: %s", target.left));
 
+        // Get to validate only; will throw IRE if option is invalid
+        CompressionParams valueCompression = getCompressionOptionChecked(options, IndexContext.VALUE_COMPRESSION_OPTION_NAME);
```

Comment:
Nit: We could put this below, after getting the key compression, and right before checking the format version. That is, immediately before its first usage.